File: Pa2_Files_1/Lex.c
//-----------------------------------------------------------------------------
// Name: Garland Simpson
// cruz-id: gfsimpso
// Assignment: pa2
//
// Lex.c
// Illustrates file input-output commands and text processing using the
// string functions strtok and strcat.
//-----------------------------------------------------------------------------




#include<stdio.h>
#include<stdlib.h>
#include<string.h>


#include "List.h"
#define MAX_LEN 160
//#pragma warning(disable: 4996) //4996 for _CRT_SECURE_NO_WARNINGS equivalent
// deprecated code here

int main(int argc, char * argv[]) {

	int i, count = 0;
	FILE *in, *out;
	char line[MAX_LEN];
	char ar[MAX_LEN][256];
	char* token;

	// check command line for correct number of arguments
	if (argc != 3) {
		printf("Usage: %s <input file> <output file>\n", argv[0]);
		exit(1);
	}

	// open files for reading and writing 
	in = fopen(argv[1], "r");
	out = fopen(argv[2], "w");
	if (in == NULL) {
		printf("Unable to open file %s for reading\n", argv[1]);
		exit(1);
	}
	if (out == NULL) {
		printf("Unable to open file %s for writing\n", argv[2]);
		exit(1);
	}

	
	/* read each line of input file, then count and print tokens */
	
	
	while (fgets(line, MAX_LEN, in) != NULL) {
		strcpy(ar[count], line);
		count++;
	}
	token = strcat(line, " \n");
	strcpy(ar[count - 1], line);
	
	

	List myList = newList();

	append(myList, 0);

	for (i = 1; i < count; i++)
	{
		for (moveFront(myList); index(myList) >= 0; moveNext(myList))
		{
			if (strcmp(ar[i], ar[get(myList)]) < 0)
			{
				insertBefore(myList, i);
				break;
			}
		}

		if (index(myList) == -1)
			append(myList, i);
	}
	for (moveFront(myList); index(myList) >= 0; moveNext(myList))
	{
		fprintf(out, "%s", ar[get(myList)]);
		
	}
	freeList(&myList);
	/* close files */
	fclose(in);
	fclose(out);


	return(0);
}

File: Pa1_Files/Lex.java
/**
 * Name: Garland Simpson
 * Cruzid: 
 * Assignment: pa1
 */
import java.io.*;
import java.util.ArrayList;
import java.util.Scanner;
import java.io.File;

public class Lex
{
    public static void main(String[] args) throws IOException
    {
        Scanner in = null;
        PrintWriter out = null;
        String line;
        ArrayList<String> token = new ArrayList<>();
        int n = 0;
        int i = 0;

        if(args.length < 2)
        {
            System.err.println("Usage: FileIO infile outfile");
            System.exit(1);
        }

        in = new Scanner(new File(args[0]));
        out = new PrintWriter(new FileWriter(args[1]));

        List myList = new List();



        while( in.hasNextLine() )
        {
            n++;
            line = in.nextLine();
            token.add(i, line);
            i++;
        }

        myList.append(0);

        for(i = 1; i < n; i++)
        {
            for(myList.moveFront(); myList.index() >= 0; myList.moveNext())
            {
                if (token.get(i).compareTo(token.get(myList.get())) < 0)
                {
                    myList.insertBefore(i);
                    break;
                }
            }

            if (myList.index() == -1)
                myList.append(i);
        }

        for(myList.moveFront(); myList.index() >= 0; myList.moveNext())
        {
            out.println(token.get(myList.get()));
        }

        in.close();
        out.close();
    }
}
